cns/restserver/restserver.go

```diff
@@ -185,6 +185,9 @@ func NewHTTPRestService(config *common.ServiceConfig, wscli interfaceGetter, wsp
 		return nil, errors.Wrap(err, "failed to get primary interface from IMDS response")
 	}
 
+	// add primaryInterfaceIP to cns config
+	config.PrimaryInterfaceIP = primaryInterface.PrimaryIP
```

Comment:
Is it required to add a new field in service config or can we refer from `serviceState.primaryInterface.PrimaryIP` ?

Reply:
yes, we need to get primary interface IP from here and then use it in AddListener(config *common.ServiceConfig)